concordium-consensus/src/Concordium/GRPC2.hs

```diff
@@ -535,11 +563,714 @@ instance ToProto AccountInfo where
+            JustForCPV1 as -> ProtoFields.parameterTime .= toProto as)
+
+-- |Is not needed if we can make a ToProto instance for HigherLevelKeys.
+toProtoKeys :: Updates.HigherLevelKeys kind -> Proto.HigherLevelKeys
+toProtoKeys keys = Proto.make $ do
+    ProtoFields.keys .= map toProto (Vec.toList $ Updates.hlkKeys keys)
+    ProtoFields.threshold .= toProto (Updates.hlkThreshold keys)
+
+-- |Is not needed if we can make a ToProto instance for Authorizations.
+toProtoAuthorizationsV0 :: Updates.Authorizations cpv -> Proto.AuthorizationsV0
+toProtoAuthorizationsV0 auth = Proto.make $ do
+    ProtoFields.keys .= map toProto (Vec.toList $ Updates.asKeys auth)
+    ProtoFields.emergency .= toProto (Updates.asEmergency auth)
+    ProtoFields.protocol .= toProto (Updates.asProtocol auth)
+    ProtoFields.parameterElectionDifficulty .= toProto (Updates.asParamElectionDifficulty auth)
+    ProtoFields.parameterEuroPerEnergy.= toProto (Updates.asParamEuroPerEnergy auth)
+    ProtoFields.parameterMicroCCDPerEuro .= toProto (Updates.asParamMicroGTUPerEuro auth)
+    ProtoFields.parameterFoundationAccount .= toProto (Updates.asParamFoundationAccount auth)
+    ProtoFields.parameterMintDistribution .= toProto (Updates.asParamMintDistribution auth)
+    ProtoFields.parameterTransactionFeeDistribution .= toProto (Updates.asParamTransactionFeeDistribution auth)
+    ProtoFields.parameterGasRewards .= toProto (Updates.asParamGASRewards auth)
+    ProtoFields.poolParameters .= toProto (Updates.asPoolParameters auth)
+    ProtoFields.addAnonymityRevoker .= toProto (Updates.asAddAnonymityRevoker auth)
+    ProtoFields.addIdentityProvider .= toProto (Updates.asAddIdentityProvider auth)
+
+instance ToProto Updates.AccessStructure where
+  type Output Updates.AccessStructure = Proto.AccessStructure
+  toProto Updates.AccessStructure{..} = Proto.make $ do
+    ProtoFields.accessPublicKeys .= map toProtoUpdateKeysIndex (Set.toList accessPublicKeys)
+    ProtoFields.accessThreshold .= toProto accessThreshold
+    where toProtoUpdateKeysIndex i = Proto.make $ ProtoFields.value .= fromIntegral i
+
+instance ToProto Updates.UpdatePublicKey where
+  type Output Updates.UpdatePublicKey = Proto.UpdatePublicKey
+  toProto = mkSerialize
+
+instance ToProto Updates.UpdateKeysThreshold where
+  type Output Updates.UpdateKeysThreshold = Proto.UpdateKeysThreshold
+  toProto Updates.UpdateKeysThreshold{..} = Proto.make $ ProtoFields.value .= fromIntegral uktTheThreshold
+
+instance ToProto MintRate where
+  type Output MintRate = Proto.MintRate
+  toProto MintRate{..} = Proto.make $ do
+    ProtoFields.mantissa .= mrMantissa
+    ProtoFields.exponent .= fromIntegral mrExponent
+
+instance ToProto Wasm.WasmVersion where
+  type Output Wasm.WasmVersion = Proto.ContractVersion
+  toProto Wasm.V0 = Proto.V0
+  toProto Wasm.V1 = Proto.V1
+
+instance ToProto Wasm.ContractEvent where
+  type Output Wasm.ContractEvent = Proto.ContractEvent
+  toProto = mkSerialize
```

Comment:
I think this should be 
```
toProto (ContractEvent short) = Proto.make $ Proto.value .= BSS.toByteString short
```
since we don't want to be duplicating the length information.